kafkabp/consumer.go

```diff
@@ -113,12 +117,16 @@ type consumer struct {
 // - If GroupID is empty, it creates a consumer that has the whole view of the
 // topic. This implementation of Kafka consumer is suitable for use cases like
 // deliver config/data through Kafka to services.
-func NewConsumer(cfg ConsumerConfig) (Consumer, error) {
+func NewConsumer(cfg ConsumerConfig, opts ...SaramaConfigOverrider) (Consumer, error) {
```

Comment:
@kylelemons technically this is a breaking change, but it should not break users' code in 99% of the cases.

if you want to be strict I can also add a `NewConsumerWithOverrider` function and keep this one intact.

Reply:
Nevermind added `NewConsumerWithConfigOverriders` instead. I think that's also easier for us to track how many services are using it.